src/test/CATALOG_FITS_VOT.test.ts

```diff
@@ -0,0 +1,273 @@
+            let CatalogFileInfoAck = await Connection.receive(CARTA.CatalogFileInfoResponse);
+            // console.log(CatalogFileInfoAck.fileInfo)
+            expect(CatalogFileInfoAck.success).toEqual(assertItem.catalogFileInfoResponse[index].success);
+            expect(CatalogFileInfoAck.fileInfo.name).toEqual(assertItem.catalogFileInfoResponse[index].fileInfo.name);
+            expect(CatalogFileInfoAck.fileInfo.fileSize.low).toEqual(assertItem.catalogFileInfoResponse[index].fileInfo.fileSize);
+            expect(CatalogFileInfoAck.fileInfo.description).toMatch(assertItem.catalogFileInfoResponse[index].fileInfo.description)
+            if (CatalogFileInfoAck.fileInfo.type) {
+                expect(CatalogFileInfoAck.fileInfo.type).toEqual(assertItem.catalogFileInfoResponse[index].fileInfo.type);
+            };
+            expect(CatalogFileInfoAck.headers.length).toEqual(assertItem.catalogFileInfoResponse[index].lengthOfHeaders);
+        });
+
+        test(`(Step 5) Request CatalogFile & check CatalogFileAck | `, async () => {
+            await Connection.send(CARTA.OpenCatalogFile, assertItem.openCatalogFile[index]);
+            let CatalogFileAck = await Connection.receive(CARTA.OpenCatalogFileAck);
+            // console.log(CatalogFileAck.fileInfo)
+            expect(CatalogFileAck.success).toEqual(assertItem.openCatalogFileAck[index].success);
+            expect(CatalogFileAck.dataSize).toEqual(assertItem.openCatalogFileAck[index].dataSize);
+            expect(CatalogFileAck.fileId).toEqual(assertItem.openCatalogFileAck[index].fileId);
+            expect(CatalogFileAck.fileInfo.name).toEqual(assertItem.openCatalogFileAck[index].fileInfo.name);
+            expect(CatalogFileAck.fileInfo.fileSize.low).toEqual(assertItem.openCatalogFileAck[index].fileInfo.fileSize);
+            if (CatalogFileAck.fileInfo.type) {
+                expect(CatalogFileAck.fileInfo.type).toEqual(assertItem.openCatalogFileAck[index].fileInfo.type);
+            };
+            expect(CatalogFileAck.headers.length).toEqual(assertItem.openCatalogFileAck[index].lengthOfHeaders);
+        }, openCatalogLargeTimeout);
+
+        test(`(Step 6) Request CatalogFilter: progress & check CatalogFilterResponse | `, async () => {
+            await Connection.send(CARTA.CatalogFilterRequest, assertItem.catalogFilterReq[index]);
+            let CatalogFilterResponse = await Connection.receive(CARTA.CatalogFilterResponse);
+            let ReceiveProgress = CatalogFilterResponse.progress
+            if (ReceiveProgress != 1) {
+                while (ReceiveProgress < 1) {
+                    CatalogFilterResponse = await Connection.receive(CARTA.CatalogFilterResponse);
+                    ReceiveProgress = CatalogFilterResponse.progress
+                    console.warn('' + assertItem.catalogFileInfoReq[index].name + ' Catalog loading progress :', ReceiveProgress)
+                };
```

Comment:
Please use 
```
let ack = Client.streamUntil( (type, data) => type == CARTA. CatalogFilterResponse && data. progress==1);
let CatalogFilterResponse = ack.CatalogFilterResponse[0];
```
to avoid the situation as receiving other type of message before progress==1.

Reply:
If we need console updating the process in time, keep the original code.